### PR TITLE
Pdct 1066 user org not checked on document delete

### PR DESCRIPTION
**Pull request by katybaulch**
# Description

Check user org matches org the document belongs to on document delete.

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [x] Skip auto-tagging
- [ ] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] GitHub workflow update
- [ ] Documentation update
- [ ] Remove legacy code
- [ ] Dependency update

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [ ] The PR represents a single feature (small drive-by fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: app/repository/document.py [hunk 1]
```diff
@@ -480,3 +480,12 @@ def count(db: Session, org_id: Optional[int]) -> Optional[int]:
...
 
     return n_documents
+
+
+def get_org_from_import_id(db: Session, import_id: str) -> Optional[int]:
+    result = _get_query(db).filter(FamilyDocument.import_id == import_id).one_or_none()
+    _LOGGER.error(result)
+    if result is None:
+        return None
+    _, _, org, _, _ = result
```
Comment (diversemix) [inline app/repository/document.py:490]: I never knew you could use `_` multiple times ... noice!

File: app/service/document.py [hunk 2]
```diff
@@ -148,16 +148,36 @@ def create(
...
+
+    doc = get(import_id)
+    if doc is None:
+        return None
+
     if context is not None:
         context.error = f"Could not delete document {import_id}"
+
+    entity_org_id = get_org_from_id(db, import_id)
+    app_user.is_authorised_to_make_changes(db, user_email, entity_org_id, import_id)
```
Comment (diversemix) [inline app/service/document.py:173]: Just a note - I wonder if the return value of this function should be removed - and the function renames to something like `raise_if_not_in_organisation` ... just that the function as-is doesn't actually return false so could be misleading - not a blocker - food for thought

**Review by diversemix — APPROVED** [review]
lgtm + :speech_balloon:

File: app/service/document.py (hunk 2 above)
Comment (katybaulch) [reply app/service/document.py:173]: Yeah I like that idea - I might implement in the latest PR to avoid a bunch of merge conflicts though :)

**Pull request merged**